packages/thirdweb/src/react/native/ui/connect/ConnectedButton.tsx

```diff
@@ -0,0 +1,184 @@
+import { useQuery } from "@tanstack/react-query";
+import { Image, StyleSheet, View } from "react-native";
+import { SvgXml } from "react-native-svg";
+import { nativeLocalStorage } from "../../../../utils/storage/nativeStorage.js";
+import { getWalletInfo } from "../../../../wallets/__generated__/getWalletInfo.js";
+import type { Account, Wallet } from "../../../../wallets/interfaces/wallet.js";
+import { parseTheme } from "../../../core/design-system/CustomThemeProvider.js";
+import type { ConnectButtonProps } from "../../../core/hooks/connection/ConnectButtonProps.js";
+import { getLastAuthProvider } from "../../../core/utils/storage.js";
+import { useConnectedWalletDetails } from "../../../core/utils/wallet.js";
+import { fontSize, spacing } from "../../design-system/index.js";
+import { useActiveWalletChain } from "../../hooks/wallets/useActiveWalletChain.js";
+import { useDisconnect } from "../../hooks/wallets/useDisconnect.js";
+import { ThemedButton } from "../components/button.js";
+import { ThemedText } from "../components/text.js";
+import {
+  APPLE_ICON,
+  EMAIL_ICON,
+  FACEBOOK_ICON,
+  GOOGLE_ICON,
+  PHONE_ICON,
+  WALLET_ICON,
+} from "../icons/svgs.js";
+
+export function ConnectedButton(
+  props: ConnectButtonProps & {
+    onClose: () => void;
+    wallet: Wallet;
+    account: Account;
+  },
+) {
+  const theme = parseTheme(props.theme);
+  const { account, wallet } = props;
+  const { disconnect } = useDisconnect();
+  const walletChain = useActiveWalletChain();
+  const { ensAvatarQuery, addressOrENS, balanceQuery } =
+    useConnectedWalletDetails(
+      props.client,
+      walletChain,
+      account,
+      props.detailsButton?.displayBalanceToken,
+    );
+  return (
+    wallet &&
```

Comment:
Why do we need to check for wallet and account here if they're required props?

Reply:
Nope, will remove